src/cdi_ndi/NDI_CP_Eloss.cc

```diff
@@ -124,6 +124,7 @@ NDI_CP_Eloss::NDI_CP_Eloss(const std::string &library_in, rtt_cdi::CParticle tar
  * see https://xweb.lanl.gov/projects/data/nuclear/ndi/ndi.html
  */
 void NDI_CP_Eloss::load_ndi() {
+#if NDI_DEDX_SUPPORT
```

Comment:
I found this more efficient than having alternate constructors if NDI support is inefficient